cli/config/credentials/file_store.go

```diff
@@ -65,6 +65,8 @@ Configure a credential helper to remove this warning. See
 https://docs.docker.com/go/credential-store/
 `
 
+var alreadyPrinted bool
```

Comment:
I don't _think_ this needs to be a thread-safe type, but let me know if you think otherwise.

Reply:
We could make it a property of the `fileStore` perhaps? Could even be a `sync.OnceFunc` that prints the warning, e.g. `c.printWarning()`

Reply:
Hmm, I worry that there's a few places initializing the store through the code, so making it a property of the store wouldn't necessarily make the warning "once per CLI invocation" but rather "once per store". I'd rather this be a package var instead (but can be a `sync.OnceFunc`). WDYT?

Reply:
also @vvoland WDYT?

Reply:
Yeah I think it's fine to do that in the package as the fileStore shouldn't really be printing out messages 🙈 But I think that we can't really move it somewhere so the caller can do it?

Reply:
We kind of had that before 😅 To be fair, it doesn't really fit well anywhere https://github.com/docker/cli/pull/5259

Reply:
Ended up changing this to an `atomic.Bool` as per https://github.com/docker/cli/pull/5344#pullrequestreview-2238001263

> but not 100% sure if (e.g.) Piñata or other codebases may be executing this code; (maybe not too problematic?)

Can't hurt to be too safe, this has bit us in the ass before.